rclpy/src/rclpy/qos.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__QOS_HPP_
+#define RCLPY__QOS_HPP_
+
+#include <rmw/qos_profiles.h>
+
+#include <pybind11/pybind11.h>
+
+#include <string>
+
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+enum class QoSCompatibility
+{
+  OK = RMW_QOS_COMPATIBILITY_OK,
+  WARNING = RMW_QOS_COMPATIBILITY_WARNING,
+  ERROR = RMW_QOS_COMPATIBILITY_ERROR,
+};
+
+/// Result type for checking QoS compatibility
+/**
+ * \see rclcpp::qos_check_compatible()
+ */
+struct QoSCheckCompatibleResult
+{
+  /// Compatibility result.
+  QoSCompatibility compatibility;
```

Comment:
Rather than defining a `QoSCompatibility` enum in this header, I would make this type [`rmw_qos_compatibility_type_t`](https://github.com/ros2/rmw/blob/ace271650ad39a656d81239050c9639c4190f077/rmw/include/rmw/qos_profiles.h#L126). Then there's no need for the enum above or the switch statement in `qos_check_compatible`